```
06:04:34 --- Skipped scenarios:
06:04:34 
06:04:34     /var/lib/jenkins/workspace/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ/tests/integration/features/auth.feature:10
06:04:34 
06:04:34 438 scenarios (437 passed, 1 skipped)
06:04:34 4109 steps (4105 passed, 4 skipped)
06:04:34 39m34.51s (27.30Mb)
06:04:35 ./run.sh: line 121:  6249 Terminated              php -S localhost:$PORT -t "$OC_PATH"
06:04:35 ./run.sh: line 121:  6250 Terminated              php -S localhost:$PORT_FED -t ../..
06:04:35 Config value enable_external_storage for app core set to no
06:04:35 No such app enabled: testing
06:04:36 {"reqId":"CbKq4RDJV2RXtLzohJYx","level":3,"time":"2017-07-05T04:02:48+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v1.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/testing\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"CbKq4RDJV2RXtLzohJYx","level":3,"time":"2017-07-05T04:02:48+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v1.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/theme-example\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"CbKq4RDJV2RXtLzohJYx","level":3,"time":"2017-07-05T04:02:48+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v1.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/comments\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"nF981MuYGjchER9pb3g1","level":3,"time":"2017-07-05T04:04:24+00:00","remoteAddr":"::1","user":"test","app":"OCP\\Share","method":"GET","url":"\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/sharees?search=test%40localhost&itemType=calendar","message":"Sharing backend for calendar not found"}
06:04:36 {"reqId":"jZ2Vt3OmbB0onqy521Nn","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/systemtags\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"jZ2Vt3OmbB0onqy521Nn","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/multidirtest\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"jZ2Vt3OmbB0onqy521Nn","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/testing\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"jZ2Vt3OmbB0onqy521Nn","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/theme-example\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"jZ2Vt3OmbB0onqy521Nn","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"::1","user":"admin","app":"PHP","method":"GET","url":"\/ocs\/v2.php\/cloud\/apps?filter=enabled","message":"file_get_contents(\/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/apps\/comments\/appinfo\/signature.json): failed to open stream: No such file or directory at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/private\/IntegrityCheck\/Helpers\/FileAccessHelper.php#39"}
06:04:36 {"reqId":"Nv75y1oCCAxi9OjavTyQ","level":3,"time":"2017-07-05T04:04:34+00:00","remoteAddr":"","user":"--","app":"PHP","method":"--","url":"--","message":"Not quoting the scalar \"%paths.base%\/..\/output\/\" starting with the \"%\" indicator character is deprecated since Symfony 3.1 and will throw a ParseException in 4.0. at \/var\/lib\/jenkins\/workspace\/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ\/lib\/composer\/symfony\/yaml\/Inline.php#325"}
06:04:36 runsh: Exit code: 1
06:04:36 Makefile:9: recipe for target 'test' failed
06:04:36 make[1]: *** [test] Error 1
06:04:36 make[1]: Leaving directory '/var/lib/jenkins/workspace/owncloud-core_core_PR-28292-JZVBMLTQWI7DVBPCVKB72DFZ7XE4KBXX7LD5TOASFW3FDBJUVLOQ/tests/integration'
06:04:36 Makefile:195: recipe for target 'test-integration' failed
06:04:36 make: *** [test-integration] Error 2
[Pipeline] }
[Pipeline] // timeout
[Pipeline] echo
06:04:36 Test execution failed: hudson.AbortException: script returned exit code 2
[Pipeline] step
06:04:36 Recording test results
[Pipeline] error
[Pipeline] }
[Pipeline] // wrap
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
```

Strange that `run.sh` returns exit code 1 despite skipped tests. Does it do so when run locally ?

--- a/tests/ui/features/bootstrap/LoginContext.php
+++ b/tests/ui/features/bootstrap/LoginContext.php
@@ -35,6 +35,7 @@ class LoginContext extends RawMinkContext implements Context
 {
 	private $loginPage;
 	private $filesPage;
+	private $expectedPage;
 	private $featureContext;
 
 	public function __construct(LoginPage $loginPage)
@@ -57,6 +58,24 @@ class LoginContext extends RawMinkContext implements Context
 	{
 		$this->filesPage = $this->loginPage->loginAs($username, $password);
 		$this->filesPage->waitTillPageIsLoaded($this->getSession());
+	}
+
+	/**
+	 * @When I login with username :username and invalid password :password
+	 */
+	public function iLoginWithUsernameAndInvalidPassword($username, $password)
+	{
+		$this->loginPage->loginAs($username, $password, 'LoginPage');
+		$this->loginPage->waitTillPageIsLoaded($this->getSession());
+	}
+
+	/**
+	 * @When I login with username :username and password :password after a redirect from the :page page
+	 */
+	public function iLoginWithUsernameAndPasswordAfterRedirectFromThePage($username, $password, $page)
+	{
+		$this->expectedPage = $this->loginPage->loginAs($username, $password, str_replace(' ', '', ucwords($page)) . 'Page');
+		$this->expectedPage->waitTillPageIsLoaded($this->getSession());
 	}
 
 	/**

--- a/tests/ui/features/bootstrap/PersonalGeneralSettingsContext.php
+++ b/tests/ui/features/bootstrap/PersonalGeneralSettingsContext.php
@@ -48,6 +48,15 @@ class PersonalGeneralSettingsContext extends RawMinkContext implements Context
 	}
 	
 	/**
+	 * @Given I go to the personal general settings page
+	 */
+	public function iGoToThePersonalGeneralSettingsPage()
+	{
+		$this->visitPath($this->personalGeneralSettingsPage->getPagePath());
+		$this->personalGeneralSettingsPage->waitForOutstandingAjaxCalls($this->getSession());
+	}
+	
+	/**
 	 * @When I change the language to :language
 	 */
 	public function iChangeTheLanguageTo($language)

--- a/tests/ui/features/lib/LoginPage.php
+++ b/tests/ui/features/lib/LoginPage.php
@@ -23,6 +23,7 @@
 namespace Page;
 
 use SensioLabs\Behat\PageObjectExtension\PageObject\Page;
+use Behat\Mink\Session;
 
 class LoginPage extends OwncloudPage
 {
@@ -30,12 +31,31 @@ class LoginPage extends OwncloudPage
 	 * @var string $path
 	 */
 	protected $path = '/index.php/login';
+	protected $userInputId = "user";
+	protected $passwordInputId = "password";
 
-	public function loginAs($username, $password)
+	public function loginAs($username, $password, $target='FilesPage')
 	{
-		$this->fillField("user", $username);
-		$this->fillField("password", $password);
+		$this->fillField($this->userInputId, $username);
+		$this->fillField($this->passwordInputId, $password);
 		$this->findById("submit")->click();
-		return $this->getPage('FilesPage');
+		return $this->getPage($target);
+	}
+
+	//there is no reliable loading indicator on the login page, so just wait for
+	//the user and password to be there.
+	public function waitTillPageIsLoaded(Session $session, $timeout_msec=STANDARDUIWAITTIMEOUTMILLISEC)
+	{
+		$currentTime = microtime(true);
+		$end = $currentTime + ($timeout_msec / 1000);
+		while ($currentTime <= $end) {
+			if (($this->findById($this->userInputId) !== null) &&
+			    ($this->findById($this->passwordInputId) !== null)) {
+				break;
+			}
+			usleep(STANDARDSLEEPTIMEMICROSEC);
+			$currentTime = microtime(true);
+		}
+		$this->waitForOutstandingAjaxCalls($session);
 	}
 }

--- a/tests/ui/features/lib/OwncloudPage.php
+++ b/tests/ui/features/lib/OwncloudPage.php
@@ -38,11 +38,13 @@ class OwncloudPage extends Page
 		$end = $currentTime + ($timeout_msec / 1000);
 		while ($currentTime <= $end) {
 			$loadingIndicator=$this->find("css", '.loading');
-			$visibility = $this->elementHasCSSValue(
-				$loadingIndicator, 'visibility', 'visible'
-			);
-			if ($visibility===FALSE) {
-				break;
+			if (!is_null($loadingIndicator)) {
+				$visibility = $this->elementHasCSSValue(
+					$loadingIndicator, 'visibility', 'visible'
+				);
+				if ($visibility===FALSE) {
+					break;
+				}
 			}
 			usleep(STANDARDSLEEPTIMEMICROSEC);
 			$currentTime = microtime(true);
@@ -116,6 +118,14 @@ class OwncloudPage extends Page
 	 */
 	public function getMyUsername() {
 		return $this->findById($this->userNameDispayId)->getText();
+	}
+
+	/**
+	 * return the path to the relevant page
+	 * @return string
+	 */
+	public function getPagePath() {
+		return $this->getPath();
 	}
 
 	/**

--- a/tests/ui/features/lib/PersonalGeneralSettingsPage.php
+++ b/tests/ui/features/lib/PersonalGeneralSettingsPage.php
@@ -23,6 +23,7 @@
 namespace Page;
 
 use SensioLabs\Behat\PageObjectExtension\PageObject\Page;
+use Behat\Mink\Session;
 
 class PersonalGeneralSettingsPage extends OwncloudPage
 {
@@ -32,9 +33,26 @@ class PersonalGeneralSettingsPage extends OwncloudPage
 	 */
 	protected $path = '/index.php/settings/personal?sectionid=general';
 	protected $languageSelectId = "languageinput";
+	protected $personalProfilePanelId = "OC\Settings\Panels\Personal\Profile";
 	
 	public function changeLanguage($language)
 	{
 		$this->selectFieldOption($this->languageSelectId, $language);
+	}
+
+	//there is no reliable loading indicator on the personal general settings page, so just wait for
+	//the personal profile panel to be there.
+	public function waitTillPageIsLoaded(Session $session, $timeout_msec=STANDARDUIWAITTIMEOUTMILLISEC)
+	{
+		$currentTime = microtime(true);
+		$end = $currentTime + ($timeout_msec / 1000);
+		while ($currentTime <= $end) {
+			if ($this->findById($this->personalProfilePanelId) !== null) {
+				break;
+			}
+			usleep(STANDARDSLEEPTIMEMICROSEC);
+			$currentTime = microtime(true);
+		}
+		$this->waitForOutstandingAjaxCalls($session);
 	}
 }

--- a/tests/ui/features/login.feature
+++ b/tests/ui/features/login.feature
@@ -10,3 +10,23 @@ Feature: login
 		Given I am on the login page
 		When I login with username "admin" and password "admin"
 		Then I should be redirected to a page with the title "Files - ownCloud"
+
+	Scenario: admin login with invalid password
+		Given I am on the login page
+		When I login with username "admin" and invalid password "invalidpassword"
+		Then I should be redirected to a page with the title "ownCloud"
+
+	Scenario: access the personal general settings page when not logged in
+		Given I go to the personal general settings page
+		Then I should be redirected to a page with the title "ownCloud"
+		When I login with username "admin" and password "admin" after a redirect from the "personal general settings" page
+		Then I should be redirected to a page with the title "Settings - ownCloud"
+
+	@skip @entering-the-wrong-password-breaks-the-redirect-issue-28129
+	Scenario: access the personal general settings page when not logged in using incorrect then correct password
+		Given I go to the personal general settings page
+		Then I should be redirected to a page with the title "ownCloud"
+		When I login with username "admin" and invalid password "qwerty"
+		Then I should be redirected to a page with the title "ownCloud"
+		When I login with username "admin" and password "admin" after a redirect from the "personal general settings" page
+		Then I should be redirected to a page with the title "Settings - ownCloud"

